app/models/choice.rb

```diff
@@ -62,8 +62,20 @@ def option_node_id
 
   # This is a temporary method for assigning option based on an OptionNode ID.
   # Eventually Options will be removed and OptionNodes will be stored on Choices directly.
+  #
+  # Raises a loud error if the OptionNode is not in the OptionSet (or the mission) for security purposes.
   def option_node_id=(id)
-    self.option_id = id.present? ? OptionNode.id_to_option_id(id) : nil
+    self.option_id = if id.present?
+                       # Answer can't always be set in advance, so only scope it if it's set.
+                       scope = if answer.present?
+                                 OptionNode.where(option_set_id: answer.option_set.id)
+                               else
+                                 OptionNode
+                               end
+                       option_id = scope.id_to_option_id(id)
+                       raise ArgumentError if option_id.nil?
+                       option_id
+                     end
```

Comment:
This assertion is new since pairing. It's required in order to validate ODK select-multiple answers, since they skirt around the call to `choices_attributes=`.

I found a seemingly safe way to pass `answer` here at least for ODK; if we come up with a way to pass it in for web as well, then we could remove the `scope` conditional here and also remove `choices_attributes=` from Answer since this would cover it.

The reason it's difficult to pass `answer` in from web is because it uses a more generic call to `tree_parent.children.build` (compared to a more explicit `answer.choices.build` from ODK).